event-log/src/main/scala/io/renku/eventlog/init/EventPayloadTableCreator.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.eventlog.init
+
+import cats.effect.Bracket
+import ch.datascience.db.{DbClient, DbTransactor}
+import ch.datascience.graph.model.events.EventStatus.GenerationRecoverableFailure
+import io.chrisdavenport.log4cats.Logger
+import io.renku.eventlog.EventLogDB
+
+private trait EventPayloadTableCreator[Interpretation[_]] {
+  def run(): Interpretation[Unit]
+}
+
+private object EventPayloadTableCreator {
+  def apply[Interpretation[_]](
+      transactor: DbTransactor[Interpretation, EventLogDB],
+      logger:     Logger[Interpretation]
+  )(implicit ME:  Bracket[Interpretation, Throwable]): EventPayloadTableCreator[Interpretation] =
+    new EventPayloadTableCreatorImpl(transactor, logger)
+}
+
+private class EventPayloadTableCreatorImpl[Interpretation[_]](
+    transactor: DbTransactor[Interpretation, EventLogDB],
+    logger:     Logger[Interpretation]
+)(implicit ME:  Bracket[Interpretation, Throwable])
+    extends EventPayloadTableCreator[Interpretation]
+    with EventTableCheck[Interpretation] {
+
+  import cats.syntax.all._
+  import doobie.implicits._
+
+  private implicit val transact: DbTransactor[Interpretation, EventLogDB] = transactor
+
+  override def run(): Interpretation[Unit] =
+    whenEventTableExists(
+      logger info "'event_payload' table creation skipped",
+      otherwise = checkTableExists flatMap {
+        case true  => logger info "'event_payload' table exists"
+        case false => createTable
+      }
+    )
+
+  private def checkTableExists: Interpretation[Boolean] =
+    sql"select event_id from event_payload limit 1"
+      .query[String]
+      .option
+      .transact(transactor.get)
+      .map(_ => true)
+      .recover { case _ => false }
+
+  private def createTable = for {
+    _ <- createTableSql.update.run transact transactor.get
+    _ <- execute(sql"CREATE INDEX IF NOT EXISTS idx_event_id ON event_payload(event_id)")
+    _ <- execute(sql"CREATE INDEX IF NOT EXISTS idx_project_id ON event_payload(project_id)")
+    _ <- logger info "'event_payload' table created"
+  } yield ()
+
+  private lazy val createTableSql = sql"""
```

Comment:
I guess it would be nice if we have a foreign key on the table added. You can maybe take a look here:
https://github.com/SwissDataScienceCenter/renku-graph/blob/2a680e9db8859152962f0ed71a4992377e6acb60/event-log/src/main/scala/io/renku/eventlog/init/ProjectTableCreator.scala#L102
However, the foreign key will be probably also compound.